conans/client/rest/conan_requester.py

```diff
@@ -61,6 +64,22 @@ def __init__(self, config, http_requester=None):
             else:
                 self._client_certificates = self._client_cert_path
 
+    def _get_retries(self, retry, retry_wait):
+        retry = retry if retry is not None else 2
+        retry_wait = retry_wait if retry_wait is not None else 5
+        if retry == 0:
+            return 0
+        retry_status_code_set = set()
+        for status in http.HTTPStatus:
+            code = int(status)
+            if code // 100 == 5:
+                retry_status_code_set.add(code)
+        return urllib3.Retry(
+            total=retry,
+            backoff_factor=(retry * retry_wait) / (2**(retry-1)),
```

Comment:
This line doesn't look pretty because urllib3 Retry doesn't support constant waits between retries.

Instead, it has backoff_factor argument and in docs it states that it will sleep for total of 
{backoff factor} * (2 ** ({number of total retries} - 1)) seconds
https://urllib3.readthedocs.io/en/latest/reference/urllib3.util.html

conan is expected to sleep for retry * retry_wait seconds, so to achieve it I set backoff factor to
retry * retry_wait / (2 ** (retry - 1) seconds.

Reply:
I would say that this is not necessary. So far, if the number of retries was >0, HttpAdapter was already applying its internal policy of retries with backoff_factor, not following the Conan ``retry_wait``. So we can simplify and leave it.

The main change is that it will retry for some more conditions than before. I am fine with this, but if I understood correctly the docs, this will not implement the current Conan "retry" functionality, which will retry a transfer that is half-way interrupted, and is only valid for initiating calls, but once some data has hit the server, it will not be retried.

I have also checked that our requests>=2.8.1 seems good and supports passing the Retry object instead of int.

Reply:
I removed backoff factor, but it looks like default value for it is zero, so there will be no backoff whatsoever. Maybe we can add some default small value like 0.1 seconds?

I thought about it when I was first writing this code and decided that conan already have a retry_wait config parameter so I choose backoff_factor with respect to it

Reply:
Yeah, maybe it is good to give the server at least some time, if it did a 500 or some resource limit hit. Lets give it a small default.